Packages/com.unity.inputsystem/InputSystem/Controls/DpadAxisControl.cs

```diff
@@ -0,0 +1,21 @@
+using UnityEngine.Experimental.Input.Layouts;
+
+namespace UnityEngine.Experimental.Input.Controls
+{
+    public class DpadAxisControl : AxisControl
+    {
+        public int component;
+
+        protected override void FinishSetup(InputDeviceBuilder builder)
+        {
+            base.FinishSetup(builder);
+            component = name == "x" ? 0 : 1;
+        }
+
+        public override unsafe float ReadUnprocessedValueFromState(void* statePtr)
+        {
+            var value = (m_Parent as DpadControl).ReadUnprocessedValueFromState(statePtr);
+            return value[component];
+        }
+    }
+}
```

Comment:
There's a complication with this I didn't think about. For various things such as actions to work correctly with controls, a control must have a proper state record. I.e. there must be a block of state memory associated with the control. Actions, for example, will monitor the memory block for changes and drive change processing based on that.